js/search.js

```diff
@@ -0,0 +1,70 @@
+document.querySelector('.searchBtn').addEventListener('click', (e) => {
+    console.log('searching');
+    const query = document.querySelector('#search-input').value;
+    const result = Store.searchBook(query);
+    UI.clearBooks();
+    UI.displayBooks(result); 
+  });
+
+  document.querySelector('.searchBtn').addEventListener('click', (e) => {
+    console.log('searching');
+    const query = document.querySelector('#search-input').value;
+    const result = Store.searchBook(query);
+    UI.clearBooks();
+    UI.displayBooks(result); 
+  });
+
+document.getElementById('book-form').addEventListener('submit', function (e) {
+    e.preventDefault();
+    // Clear form fields after submission
+    document.getElementById('title').value = '';
+    document.getElementById('author').value = '';
+    document.getElementById('isbn').value = '';
+});
+
+function generateRandomAlphabeticString(length) {
+    let result = '';
+    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+
+    for (let i = 0; i < length; i++) {
+        const randomIndex = Math.floor(Math.random() * characters.length);
+        result += characters.charAt(randomIndex);
+    }
+
+    return result;
+}
+
+document.getElementById('addMockBooks').addEventListener('click', function () {
+    // Call a function to add 1000 mock books
+    addMockBooks();
+});
+
+
+function addMockBooks() {
+
+    let books = [];
+
+    for (let i = 1; i <= 1000; i++) {
```

Comment:
Browsers perform page display operations in a single thread, so running long operation is not recommended, this can be handled in async calls where adding moc books will be done in background.